pegasus/helpers/hoc_helpers.rb

```diff
@@ -115,13 +115,14 @@ def complete_tutorial(tutorial={})
         weight: weight
       )
     end
+
     destination = "http://#{row[:referer]}/congrats?i=#{row[:session]}"
     destination += "&co=#{row[:company]}" unless row[:company].blank?
     destination += "&s=#{Base64.urlsafe_encode64(tutorial[:code])}" unless tutorial[:code].blank?
   end
 
   dont_cache
-  redirect (destination || "/congrats?s=#{Base64.urlsafe_encode64(tutorial[:code])}")
+  redirect CDO.studio_url("/congrats?s=#{Base64.urlsafe_encode64(tutorial[:code])}", CDO.default_scheme)
 end
 
```

Comment:
I'm not entirely clear on what the extra information in the destination is being used for, but I took it out temporarily to test that the redirect to new /congrats was getting the information it needed. It would be awesome if someone who knows more about this, especially about sampled session and weights etc, could help me walk through what exactly is happening here.

Reply:
A bit of background on sampling is that originally, we created a DB row for every time someone started a hoc script. this powered the big counter on our homepage showing # of people served. then we started sampling at a variable rate, and inferring from the rate and the # of samples how many people we served, with far fewer db accesses. however we kept recording unsampled data for companies and cartoon network, to make sure we show the right thing to those paritcipants on the congrats page.